src/main/kotlin/graphics/scenery/primitives/Atmosphere.kt

```diff
@@ -12,56 +12,73 @@ import java.lang.Math.toRadians
- * @param initSunDir [Vector3f] of the sun position. Defaults to sun elevation of the current local time.
- * @param emissionStrength Emission strength of the atmosphere shader. Defaults to 0.3f.
+ * @param initSunDirection [Vector3f] of the sun position. Defaults to sun elevation of the current local time.
+ * @param emissionStrength Emission strength of the atmosphere shader. Defaults to 1f.
  * @param latitude Latitude of the user; needed to calculate the local sun position. Defaults to 50.0, which is central Germany.
  */
-open class Atmosphere(initSunDir: Vector3f? = null, emissionStrength: Float = 0.3f, var latitude: Double = 50.0) :
+open class Atmosphere(
+    initSunDirection: Vector3f? = null,
+    emissionStrength: Float = 1.0f,
+    var latitude: Float = 50.0f
+) :
     Icosphere(10f, 2, insideNormals = true) {
 
     @ShaderProperty
-    var sunDir: Vector3f
+    var sunDirection = Vector3f(1f, 1f, 1f)
 
+    /** Is set to true if the user manually moved the sun direction. This disables automatic updating.*/
     private var sunDirectionManual: Boolean = false
+    /** Flag that tracks whether the sun position controls are currently attached to the input handler. */
+    var hasControls: Boolean = false
+        private set
+    // Coroutine job for updating the sun direction
+    private var job = CoroutineScope(Dispatchers.Default).launch(start = CoroutineStart.LAZY) {
+        logger.debug("Launched sun updating job")
+        while (this.coroutineContext.isActive) {
+            if (!sunDirectionManual) setSunDirectionFromTime()
+            delay(2.seconds)
+        }
+    }
+
+    var emissionStrength = emissionStrength
+        set(value) {
+            field = value
+            material { emissive = Vector4f(0f, 0f, 0f, emissionStrength * 0.3f) }
+        }
 
     init {
         this.name = "Atmosphere"
         setMaterial(ShaderMaterial.fromClass(this::class.java))
         material {
             cullingMode = Material.CullingMode.Front
             depthTest = Material.DepthTest.LessEqual
-            emissive = Vector4f(0f, 0f, 0f, emissionStrength)
+            emissive = Vector4f(0f, 0f, 0f, emissionStrength * 0.3f)
         }
 
         // Only use time-based elevation when the formal parameter is empty
-        if (initSunDir == null) {
-            sunDir = getSunDirFromTime()
+        if (initSunDirection == null) {
+            setSunDirectionFromTime()
         }
         else {
-            sunDir = initSunDir
+            sunDirection = initSunDirection
             sunDirectionManual = true
         }
 
         // Spawn a coroutine to update the sun direction
-        val job = CoroutineScope(Dispatchers.Default).launch {
-            while (!sunDirectionManual) {
-                sunDir = getSunDirFromTime()
-                // Wait 30 seconds
-                delay(30 * 1000)
-            }
-        }
+        job.start()
```

Comment:
If I understand this correctly, you can't change the behaviour during runtime, right? You need to decide upfront if the sun should move with the time, or be fixed? Could we make this more flexible?

Reply:
The job is always launched and it only animates the sun if the flag `sunDirectionManual` is set to false. This was easier than to re-launch the job in a new scope every time. (Maybe it should be renamed to `isSunDirectionManual`?)

The current behavior is this:
- per default, there are no controls attached and the sun moves with time (unless a direction is passed explicitly, which does not happen in the context of sciview)
- the user attaches the controls. This now enables the user to move the sun with the arrow keys. But until they actually moved it, the sun animation continues
- as soon as they moved the sun with the arrow keys, the animation is interrupted
- the flag is set to false again as soon as the controls are detached (= the checkbox is toggled off) and then the animation continues. This will cause the sun to jump back to the position calculated from the current time and latitude.

I can write a summary of this behavior into a comment to make it clear.